rasa/nlu/training_data/entities_parser.py

```diff
@@ -0,0 +1,149 @@
+# regex for: `[entity_text]((entity_type(:entity_synonym)?)|{entity_dict})`
+import re
+from json import JSONDecodeError
+from typing import Text, List, Dict, Match, Optional, NamedTuple
+
+from rasa.nlu.utils import build_entity
+
+from rasa.constants import DOCS_URL_TRAINING_DATA_NLU
+
+from rasa.nlu.constants import (
+    ENTITY_ATTRIBUTE_GROUP,
+    ENTITY_ATTRIBUTE_TYPE,
+    ENTITY_ATTRIBUTE_ROLE,
+    ENTITY_ATTRIBUTE_VALUE,
+)
+
+from rasa.utils.common import raise_warning
+
+GROUP_ENTITY_VALUE = "value"
+GROUP_ENTITY_TYPE = "entity"
+GROUP_ENTITY_DICT = "entity_dict"
+GROUP_ENTITY_TEXT = "entity_text"
+GROUP_COMPLETE_MATCH = 0
+
+# regex for: `[entity_text]((entity_type(:entity_synonym)?)|{entity_dict})`
+ENTITY_REGEX = re.compile(
+    r"\[(?P<entity_text>[^\]]+?)\](\((?P<entity>[^:)]+?)(?:\:(?P<value>[^)]+))?\)|\{(?P<entity_dict>[^}]+?)\})"
+)
+
+
+class EntityAttributes(NamedTuple):
+    """Attributes of an entity defined in the markdown data."""
+
+    type: Text
+    value: Text
+    text: Text
+    group: Optional[Text]
+    role: Optional[Text]
+
+
+class EntitiesParser:
```

Comment:
I'm using the class `EntitiesParser` as an umbrella for all the entity related processing to make it clearer later in the code. The alternative would be to import individual functions - that looks a little clumsy to me. Wdyt?

Reply:
this would still be its own module, and the way you call the functions wouldn't change. essentially
```python
import rasa.nlu.training_data.entities_parser as entities_parser
```
and call functions like 
```python
entities_parser.find_entities_in_training_example(example)
```